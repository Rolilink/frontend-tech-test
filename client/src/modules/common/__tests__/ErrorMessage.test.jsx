import React from 'react';
import { shallow } from 'enzyme';
import serializer from 'enzyme-to-json/serializer';

import ErrorMessage from '../ErrorMessage';

expect.addSnapshotSerializer(serializer);
const renderMessage = () => <div className="message" />;

describe('Common.ErrorMessage', () => {
  it('should render children when error is not passed', () => {
    const children = <div className="children" />;

    const errorMessage = shallow(
      <ErrorMessage renderMessage={renderMessage}>
        {children}
      </ErrorMessage>,
    );

    expect(errorMessage).toMatchSnapshot();
  });

  it('should render a message with renderMessage when there is an error', () => {
    const children = <div className="children" />;

    const errorMessage = shallow(
      <ErrorMessage renderMessage={renderMessage} error="error">
        {children}
      </ErrorMessage>,
    );

    expect(errorMessage).toMatchSnapshot();
  });
});
