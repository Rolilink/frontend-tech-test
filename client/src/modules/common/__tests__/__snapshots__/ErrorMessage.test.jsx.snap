// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Common.ErrorMessage should render a message with renderMessage when there is an error 1`] = `
<div
  className="ErrorMessage "
>
  <div
    className="ErrorMessage-Sign"
  >
    Here goes an error sign
  </div>
  <div
    className="ErrorMessage-Message"
  >
    <div
      className="message"
    />
  </div>
</div>
`;

exports[`Common.ErrorMessage should render children when error is not passed 1`] = `
<div
  className="children"
/>
`;
